### PR TITLE
[GEOS-10404]-OGCAPI Support `filter-crs` query parameter

### PR DESCRIPTION
**Pull request by sebastianfrey**
[GEOS-10404](https://osgeo-org.atlassian.net/browse/GEOS-10404)

This PR adds support for the `filter-crs` query parameter for `OGC API - Features` service.

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geoserver.org/latest/en/developer/policies/committing.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [x] All the build checks are green ([see automated QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html)).

For core and extension modules:

- [ ] New unit tests have been added covering the changes.
- [ ] [Documentation](https://github.com/geoserver/geoserver/tree/main/doc/en/user/source) has been updated (if change is visible to end users).
- [ ] The [REST API docs](https://github.com/geoserver/geoserver/tree/main/doc/en/api/1.0.0) have been updated (when changing configuration objects or the REST controllers).
- [ ] There is an issue in the [GeoServer Jira](https://osgeo-org.atlassian.net/browse/GEOS/summary) (except for changes that do not affect administrators or end users in any way).
- [ ] Commit message(s) must be in the form ``[GEOS-XYZWV] Title of the Jira ticket``.
- [ ] Bug fixes and small new features are presented as a single commit.
- [ ] Each commit has a single objective (if there are multiple commits, each has a separate JIRA ticket describing its goal).

### CODE REVIEW

**Comment by sebastianfrey** [comment]
I have not signed the Contribution License Agreement, I will do this later.

**Comment by sebastianfrey** [comment]
@aaime Maybe you can help me out. =) I have signed the CLA and send it to info@osgeo.org. How and where must I notify the developer list.

**Comment by aaime** [comment]
@sebastianfrey this message is enough. Now we wait for the OSGeo secretary to update the CLAs database with your entry (often takes a few days).

**Comment by sebastianfrey** [comment]
> @sebastianfrey this message is enough. Now we wait for the OSGeo secretary to update the CLAs database with your entry (often takes a few days).

@aaime Perfect. Thanks for yourr reply.

File: src/community/ogcapi/ogcapi-core/src/main/java/org/geoserver/ogcapi/APIFilterParser.java [hunk 1]
```diff
@@ -75,9 +91,21 @@ public Filter parse(String filter, String filterLang) {
...
             // the native CRS in EPSG axis order instead, best making the CRS explicit instead
             if (parsedFilter != null) {
-                DefaultCRSFilterVisitor crsDefaulter =
-                        new DefaultCRSFilterVisitor(FF, DefaultGeographicCRS.WGS84);
-                return (Filter) parsedFilter.accept(crsDefaulter, null);
+                DefaultCRSFilterVisitor crsDefaulter = new DefaultCRSFilterVisitor(FF, filterCRS);
+                parsedFilter = (Filter) parsedFilter.accept(crsDefaulter, null);
+
+                // if there is not FeatureType provided, we can not reproject
+                if (ft != null) {
```
Comment (aaime) [inline src/community/ogcapi/ogcapi-core/src/main/java/org/geoserver/ogcapi/APIFilterParser.java:98]: I believe this is actually redundant, every feature source returned by GeoServer is wrapped by GeoServerFeatureSource which, among other things, takes care of reprojecting to native:
https://github.com/geoserver/geoserver/blob/9e40cbbccb9626b8d08516c8eacaf5a8f168189b/src/main/src/main/java/org/vfny/geoserver/global/GeoServerFeatureSource.java#L398

File: src/community/ogcapi/ogcapi-features/src/main/java/org/geoserver/ogcapi/features/FeatureService.java [hunk 2]
```diff
@@ -324,8 +336,16 @@ public FeaturesResponse items(
         if (itemId != null) {
             filters.add(FF.id(FF.featureId(itemId)));
         }
+
         if (filter != null) {
-            Filter parsedFilter = filterParser.parse(filter, filterLanguage);
+            CoordinateReferenceSystem queryCRS = DefaultGeographicCRS.WGS84;
```
Comment (aaime) [inline src/community/ogcapi/ogcapi-features/src/main/java/org/geoserver/ogcapi/features/FeatureService.java:341]: I'd move this behavior in the FilterParser, there are other places where we can get a CRS as a string (e.g. STAC), better centralize it all.

Comment (sebastianfrey) [reply src/community/ogcapi/ogcapi-features/src/main/java/org/geoserver/ogcapi/features/FeatureService.java]: Moved to the `APIFilterParser`.

**Comment by sebastianfrey** [comment]
@aaime can you approve the workflows again? The first PR run has failed.

**Review by aaime — APPROVED** [review]

**Comment by sebastianfrey** [comment]
A question: What would be the procedure to back port this enhancement into the stable branch? Just creating another PR after this one got merged? Or do I have to look for something else?

**Comment by aaime** [comment]
@sebastianfrey yep, cherry-pick the commit back on the stable branch.

**Comment by aaime** [comment]
I still did not get an update from OSGeo on the CLA (even after asking directly), but the change is not big, I'm going to trust you on that and merge.

**Comment by sebastianfrey** [comment]
Thanks for your trust. If you want, I can mail you the CLA?

**Pull request merged**